src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -775,18 +752,26 @@ public override async Task SendResponseAsync(HttpStatusCode statusCode = HttpSta
                         {
                             contentLength = int.Parse(headerData.Value);
                         }
+                        else if (headerData.Name.Equals("Transfer-Encoding", StringComparison.OrdinalIgnoreCase) && headerData.Value.Equals("chunked", StringComparison.OrdinalIgnoreCase))
+                        {
+                            isChunked = true;
+                        }
                     }
                 }
 
-                if (contentLength < 0)
+                if (!_headersSent)
```

Comment:
This seems dangerous. What's the need for this?

Reply:
there are tests sending incomplete headers and this function can possibly be called multiple times. Only the first call should prepend status to the headers. .

Reply:
Where and why is it called multiple times?

It seems bad to store this on the connection. Won't this screw things up for the next request?